src/delta_lake/delta_service.py

```diff
@@ -33,6 +35,57 @@
     "vacuum",
 }
 
+DISALLOW_SQL_META_CHARS = {
+    "--",
+    "/*",
+    "*/",
+    ";",
+    "\\",
+}
+
+ALLOWED_STATEMENTS = {
+    "select",
+}
+
+FORBIDDEN_POSTGRESQL_SCHEMAS = {
+    "pg_",
+    "pg_catalog",
+    "information_schema",
+}
+
+
+def _check_query_is_valid(query: str) -> bool:
+    """
+    Check if a query is valid.
+    """
+
+    statements = sqlparse.parse(query)
+    if len(statements) != 1:
+        raise SparkQueryError(f"Query {query} must contain exactly one statement")
+
+    statement = statements[0]
+    if statement.get_type().lower() not in ALLOWED_STATEMENTS:
+        raise SparkQueryError(
+            f"Query {query} must be one of the following: {', '.join(ALLOWED_STATEMENTS)}, got {statement.get_type()}"
+        )
+
+    if any(schema in query.lower() for schema in FORBIDDEN_POSTGRESQL_SCHEMAS):
+        raise SparkQueryError(
+            f"Query {query} contains forbidden PostgreSQL schema: {', '.join(FORBIDDEN_POSTGRESQL_SCHEMAS)}"
+        )
```

Comment:
These could also cause false positives but seem much less likely to do so than the forbidden keywords (`jpg_file` is one possibility). Is there no way to get the schema from the parsed sql?

If you can get the table name(s) from the parsed sql checking those directly as prefixes seems like it'd work

Reply:
Added note, good enough for now